### PR TITLE
Add Jetpack Plugin Review Prompt to Restore Flow

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Add review prompt to restore process on the `in progress` and `finished` steps

Untitled 4

#### Testing instructions

* TBD

### CODE REVIEW

File: client/my-sites/backup/rewind-flow/review-prompt.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	const hasReceivedRemotePreferences = useSelector( ( state ) =>
+		getHasReceivedRemotePreferences( state )
+	);
+	const isDismissed = useSelector( ( state ) => getIsDismissed( state ) );
+	const hasBeenDismissedOnce = useSelector( ( state ) => getHasBeenDismissedOnce( state ) );
+
+	const dismiss = useCallback( () => {
+		dispatch( hasBeenDismissedOnce ? dismissReviewPromptPermanently() : dismissReviewPrompt() );
+	}, [ dispatch, hasBeenDismissedOnce ] );
```
Comment (elliottprogrammer) [inline client/my-sites/backup/rewind-flow/review-prompt.tsx:37]: I'm trying to figure out the logic here..  If the user clicks the "Leave Jetpack a review" a second time, the review prompt is dismissed permanently?  Can you clarify?

**Review by elliottprogrammer — COMMENTED** [review]
This is looking really great @cbauerman ! 👍

File: client/my-sites/backup/rewind-flow/review-prompt.tsx (hunk 1 above)
Comment (cbauerman) [reply client/my-sites/backup/rewind-flow/review-prompt.tsx:37]: yeah, I [copied this from some previous work](https://github.com/Automattic/wp-calypso/pull/22735) without enhancing how clear it is:

1. User is shown prompt for the first time
2. User dismisses prompt
3. Preference is set to ms unix time of dismiss
4. If the user goes to restore and 2 weeks have passed review prompt is shown again
5. Dismissing it a second time permanently hides it, never to be shown again ( preference is set to `permanent` )

I am not sure if this is the right approach, but it seems to me we should not constantly badger the people with review prompts.

**Comment by cbauerman** [comment]
⚠️ Branch re-based againt `trunk` to try to fix test failures ⚠️

File: client/my-sites/backup/rewind-flow/review-prompt.tsx (hunk 1 above)
Comment (elliottprogrammer) [reply client/my-sites/backup/rewind-flow/review-prompt.tsx:37]: Awesome, thanks for the clarification. 👍  Yes, I agree we should not continually badger the user to submit a review.

**Pull request closed without merging**